### PR TITLE
Anagolay project Idiyanale phase 1 (Nr. 1)

### PR DESCRIPTION
**Pull request by woss**
### Project Abstract

> Please provide a brief description of your project here summarising key points (1-2 paragraphs).
>

Anagolay is a peer-to-peer network that stores records of Rights (Copyright, Licenses, and Ownership), Restrictions, and Proofs of any digital content. It empowers the users to store, claim, sell, and rent their work with the correct transfer of Rights and usage of Licenses. Actual digital data is _never_ stored on the chain, only the respective cryptographic proof. As such, it acts as an identifier, verifiable by users who have access to that same data without disclosing it in the process.

> If your application is a follow-up to a previous grant, please mention which one in the first line of the abstract and include a link to previous pull requests if applicable.

We applied for a grant in 2020 and got approved under the name [Sensio Network](https://github.com/w3f/Grants-Program/pull/5). Since then we did a lot of research and improved the approach. We did rebranding; SensioPhoto is [Kelp Digital](http://Kelp.Digital) and SensioNetwork is [Anagolay Network](http://Anagolay.Network).

### For which grant [level](https://github.com/w3f/Grants-Program#level_slider-levels) are you applying? 
- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $50,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for > $100k Web3 Foundation Council approval)

### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied, renamed ( `project_name.md`) and updated.
- [x] A BTC or Ethereum (DAI/USDT) address for the payment of the milestones is provided inside the application.
- [x] I have read and acknowledged the [terms and conditions](https://github.com/w3f/Grants-Program/blob/master/docs/T&Cs.md).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process).

### CODE REVIEW

**Comment by semuelle** [comment]
Hi Daniel, thank you for your application. We will look into it as soon as possible.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the application and sorry for the late reply. I have a few questions: It currently says “Manifest is ......”. Is this on purpose? Regarding multihash, are you referring to https://multiformats.io/multihash/? Do you already have some initial mockups regarding the Demo web app? How are you planning to integrate Filecoin? Are you aware of https://github.com/ChainSafe/filecoindot?

**Comment by woss** [comment]
@Noc2 

>  I have a few questions: It currently says “Manifest is ......”. Is this on purpose?

Nope, that was 100% accidental and I'll update it today, I was sure I finished that part :) Thanks for pointing it out.

---

> Regarding multihash, are you referring to https://multiformats.io/multihash/? 

Yes, more accurately the [official rust version](https://github.com/multiformats/rust-multihash)

---

> How are you planning to integrate Filecoin? 

The plan is to have the Operation source code (Operation Version) as resilient as possible. By that I mean is, once it's stored on the IPFS it's close to impossible to un-store them( delete, remove, hunt for existing copies, etc ... ). Any technology that can provide will be considered and implemented. For now, we are going to store all the Operations on the public IPFS cluster we host, and on Filecoin using the [web3.storage](https://web3.storage). If there is a better way, I am open to suggestions. We do not want to run our Filecoin node, well, at least not yet. It requires quite beefy machines :) 

---

>  Are you aware of https://github.com/ChainSafe/filecoindot?

Yes, the reason why we are not using it ATM is that I would like to create something that works, then later we can improve and add additional layers for Filecoin, or Crust Network.

**Comment by woss** [comment]
@Noc2 description updated

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the update. Regarding “it's close to impossible to un-store them”, I guess if you don’t have an incentive for keeping files stored, they might disappear in the long run. Also running your own cluster is highly centralized. In general, I know that decentralized storage is a very difficult problem, but it would be nice if you at least work on solving this as part of the grant. Maybe include a research report regarding the potential existing solutions for decentralized storage and their advantages and disadvantages as part of milestone 1 or 2. 
Regarding milestone 3, we usually require teams to have at least some mockups for the UI. If you don’t have any at this stage, I simply suggest removing milestone 3 for now and we sign potentially a follow-up grant in the future.

**Comment by woss** [comment]
@Noc2

> Regarding “it's close to impossible to un-store them”, I guess if you don’t have an incentive for keeping files stored, they might disappear in the long run. Also running your own cluster is highly centralized. In general, I know that decentralized storage is a very difficult problem, but it would be nice if you at least work on solving this as part of the grant. Maybe include a research report regarding the potential existing solutions for decentralized storage and their advantages and disadvantages as part of milestone 1 or 2.

I don't know exactly what you want me to do here, especially the part when you mention milestones. I've updated the grant where I outline how we are planning to use decentralized storage. I don't see why any of the milestones need to be changed.


> Regarding milestone 3, we usually require teams to have at least some mockups for the UI. If you don’t have any at this stage, I simply suggest removing milestone 3 for now and we sign potentially a follow-up grant in the future.

I provided a UI mock, hopefully that will be enough.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the reply. Are you aware that Subspace Labs Acqui-Hired canyon network? See https://medium.com/subspace-network/subspace-labs-acqui-hires-team-from-canyon-network-to-bring-decentralized-storage-to-polkadot-89ad488964d5 So you probably won’t be able to use Canyon. Apart from this, I have one last question: would you be willing to reduce the price a little bit? This way it will be easier to accept by the rest of the committee.

**Comment by woss** [comment]
Thanks for the quick reply. 

> Are you aware that Subspace Labs Acqui-Hired canyon network? See https://medium.com/subspace-network/subspace-labs-acqui-hires-team-from-canyon-network-to-bring-decentralized-storage-to-polkadot-89ad488964d5 So you probably won’t be able to use Canyon.  

Nope, I didn't know that, and it's funny that it doesn't show up in DDG for the first 20 results. This changes only the future evaluations for the permanent storage, but not the current implementations. 

> ... would you be willing to reduce the price a little bit? This way it will be easier to accept by the rest of the committee.

It depends on what you define by `little bit`.

**Review by Noc2 — CHANGES_REQUESTED** [review]
It’s hard to define, since I also don’t know what the rest of the committee thinks is reasonable for this kind of work. But I personally think that the grant at the moment is a little bit expensive, also given that the deliveries are mostly useful for your own project and grant teams that working on potential start-ups ideally shouldn’t cover 100% of their initial funding via grants.

**Comment by woss** [comment]
Thank you for the reply.

> ... also given that the deliveries are mostly useful for your own project  ...

by implementing the proposed logic in Substrate, it's inherently available to the entire ecosystem, including all the existing and still emerging NFT projects to which it's highly relevant as it solves the obvious shortcomings of the current approach. Here is the example I wrote in the [grant](https://github.com/w3f/Grants-Program/blob/7d79fc8d10ffde163d39661a2303279a3ed6dfcf/applications/anagolay-project-idiyanale-phase-1.md#example-of-how-this-approach-affects-other-things).

Besides, Anagolay's implementation will lay the ground for the whole new type of applications in the creators and sharing economy, making the Substrate highly relevant outside of the DeFi domain and bringing new users to the ecosystem.

> ... and grant teams that working on potential start-ups ideally shouldn’t cover 100% of their initial funding via grants.

The project as a whole has started in 2019 and was mostly self-funded since then. The total investment to the date is 280k EUR with which we have created a proof of concept and a verified use-case (first of many) and did a lot of research in both tech and market domains. The aforementioned use-case for the photography market - [Kelp.Digital](https://kelp.digital/) - got validation and support from the European Union’s Horizon 2020 research innovation program NGI TRUST and was selected by [Media Motor Europe](https://mediamotoreurope.eu/the-numbers-behind-mmes-final-open-call-winners/) among 20 high-potential European deep-tech startups in the media sector. The PoC demo you can see [here](https://www.youtube.com/watch?v=U0t92YNaUmE).

**Review by Noc2 — COMMENTED** [review]
Thanks for the reply. In this case, I will share the application with the rest of the team. However, I have to say that I’m not fully convinced of it at the moment. But that doesn’t mean that the rest of the committee won’t approve it.

**Comment by woss** [comment]
thank you for passing the application forward and sorry for not replying earlier, had a pretty crazy day yesterday at IGF with our presentation.  Looking forward to seeing what the rest of the committee thinks about this.

File: applications/anagolay-project-idiyanale-phase-1.md [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+|     4. | Demo web app                   | Simple web app which will use the Workflow automatic execution and produce correct Proofs. Tech used Svelte or React with Typescript and Bulma for CSS. UI mockup is [here](https://bafybeib3sisk6hoejqm4ki46m6jtnzris6psrlrpqcldayorupwuqtcuui.ipfs.dweb.link/Home-page.png) and [here](https://bafybeigbep6y37ghbgmmwxtr5d2jbxi2t7rtyb3m7d3qvxaoxzve73fcry.ipfs.dweb.link/page-image-uploaded.png) |
+
+
+
+> __NOTE__: All the apps, _Nodejs_, and _Web app_ when executed must produce the same Proofs. WHY? Because if they use the same data and the same Workflow they must produce the same output. Same Workflow means using the same Operation manifest which means using the same Operation Version. 100% the same code execution!
+> 
+### Workflow: execution
+Execution of the Workflow manifest created in [#3](#workflow-manifest-generation). We will implement basic recursive and automatic execution of the Workflow only for `SYSTEM` Operations. The execution will load all the dependencies and execute them in the correct order.  We will NOT implement any kind of optimizations like caching or memoization to gain a boost in speed.
```
Comment (semuelle) [inline applications/anagolay-project-idiyanale-phase-1.md:329]: The link is broken, could you fix it? Otherwise it's hard to follow what is meant.

**Review by semuelle — DISMISSED** [review]
I think that, if enough people publish code on IPFS and enough people depend on it, there will also be automation built around fetching the latest version and updating automatically (re: npm), which means the issue of bad actors is not solved. Still, I think this is an interesting project, and the price is, I believe, what the missing part of your previous grant was.

File: applications/anagolay-project-idiyanale-phase-1.md (hunk 1 above)
Comment (woss) [reply applications/anagolay-project-idiyanale-phase-1.md]: thanks, fixed now.

**Review by semuelle — DISMISSED** [review]

**Comment by woss** [comment]
Thanks for the reply

> I think that, if enough people publish code on IPFS and enough people depend on it, there will also be automation built around fetching the latest version and updating automatically (re: npm), which means the issue of bad actors is not solved. 

I've updated the description on the `Operation Version`, maybe that wasn't clear enough.  About updating the deps, I don't think NPM as a tool will be willing to implement our way of handling the deps, but pnpm might. That doesn't mean the bad actor thing is still there. Since each operation version goes through the verification process it is not in developers' best interest to spam and cheat on the code quality. 

The verification process is not covered by this grant but here is the WIP diagram of what we are thinking about 
image

**Comment by mmagician** [comment]
The experience of your team with PolkaMusic is unfortunately rather a negative sign (you can check my [evaluation](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/polkamusic_1_mmagician.md) of their milestone) and I would be thus wary of the final outcome of your project - that's of course just personal experience.
Furthermore, I find it hard to imagine using the tech for anything practical - again, my personal opinion.

Overall, I am not convinced by the application and since we haven't seen any interest from other Grant Committee members within the last 2 weeks, I will close it, as per our standard policies.

**Comment by woss** [comment]
> The experience of your team with PolkaMusic is unfortunately rather a negative sign (you can check my evaluation of their milestone) and I would be thus wary of the final outcome of your project - that's of course just personal experience.

@mmagician I'm genuinely sorry to hear about your bad experience with PolkaMusic but then again there was just one junior from their team which we decided to let go week ago and i didn't update the grant since I didn't think you guys will do anything during the Xmass holidays. The update also reduces the cost and the milestones. 
> Furthermore, I find it hard to imagine using the tech for anything practical - again, my personal opinion.

How come? This is essentially the base for the proof generation off-chain and on-chain(if needed). Also, it is a base for the Rights Management and Licensing of multimedia. The reason why you don't see a big picture is that it is not obvious from this grant, and this is intentional since the grant covers only a small part if you wish to know everything about this let me know and I will be glad to explain you or provide with documentation which is still in the R&D stage and not published.

> Overall, I am not convinced by the application and since we haven't seen any interest from other Grant Committee members within the last 2 weeks, I will close it, as per our standard policies.


I really don't understand the lack of interest. I've spent days reading the applications that are less attractive and solve nothing of the real-world problems which you (plural) choose to finance, whereas us are trying to fix the copyrights and ownerships of digital media so ALL creators can benefit. How that is not interesting to you I cannot understand.


I suggest we reopen this PR and i commit updated grant proposal. Then you can review it again and make your decision. If you want me to explain the big picture (which is not part of the grant) let me know and we can set up the call.

**Comment by mmagician** [comment]
Ok, you can let us know once you have updated the proposal.

**Comment by woss** [comment]
@mmagician proposal updated.  I hope it will be satisfactory and enough for you to make a decision. I would like to remind you, if you would like to know more about the project ( more than this grant tells ) let me know.

**Comment by mmagician** [comment]
Thanks for the update. Please bear in mind, that as my colleagues have mentioned, the decision to award a grant is made by a committee and for your price level you need 3 approvals. Everyone will have a different perspective, background and priorities when making the decision.
Also thanks for the opportunity to learn more. Normally the application should be sufficient and self contained, but we will let you know if we have further doubts.
Exceptionally I am re-opening this PR - we can allow an additional 2 weeks to gather the necessary approvals due to the holiday season.

I will try to review it again in the coming days.

**Review by Noc2 — APPROVED** [review]
I’m happy to go ahead with it at this price. Sorry for the late reply here.

**Review by mmagician — COMMENTED** [review]
I still have my doubts regarding practicality of your solution, even after I've browser through your [docs](https://anagolay.dev/), which I assume contains at least some portion of the R&D you mentioned.

That said, I am not the one bringing this solution to market. At the current reduced price, I am willing to support the application - I can see you've done the necessary research regarding substrate and aim for decentralisation as much as possible.

**Review by mmagician — APPROVED** [review]

**Review by BenWhiteJam — APPROVED** [review]
